rllib/agents/slateq/slateq.py

```diff
@@ -0,0 +1,220 @@
+    "exploration_config": {
+        # The Exploration class to use.
+        "type": "EpsilonGreedy",
+        # Config for the Exploration class' constructor:
+        "initial_epsilon": 1.0,
+        "final_epsilon": 0.02,
+        "epsilon_timesteps": 10000,  # Timesteps over which to anneal epsilon.
+    },
+    # Switch to greedy actions in evaluation workers.
+    "evaluation_config": {
+        "explore": False,
+    },
+
+    # Minimum env steps to optimize for per train call. This value does
+    # not affect learning, only the length of iterations.
+    "timesteps_per_iteration": 1000,
+    # === Replay buffer ===
+    # Size of the replay buffer. Note that if async_updates is set, then
+    # each worker will have a replay buffer of this size.
+    "buffer_size": 50000,
+    # Whether to LZ4 compress observations
+    "compress_observations": False,
+    # If set, this will fix the ratio of replayed from a buffer and learned on
+    # timesteps to sampled from an environment and stored in the replay buffer
+    # timesteps. Otherwise, the replay will proceed at the native ratio
+    # determined by (train_batch_size / rollout_fragment_length).
+    "training_intensity": None,
+
+    # === Optimization ===
+    # Learning rate for adam optimizer for the user choice model
+    "lr_choice_model": 1e-2,
+    # Learning rate for adam optimizer for the q model
+    "lr_q_model": 1e-2,
+    # Adam epsilon hyper parameter
+    "adam_epsilon": 1e-8,
+    # If not None, clip gradients during optimization at this value
+    "grad_clip": 40,
+    # How many steps of the model to sample before learning starts.
+    "learning_starts": 1000,
+    # Update the replay buffer with this many samples at once. Note that
+    # this setting applies per-worker if num_workers > 1.
+    "rollout_fragment_length": 1000,
+    # Size of a batch sampled from replay buffer for training. Note that
+    # if async_updates is set, then each worker returns gradients for a
+    # batch of this size.
+    "train_batch_size": 32,
+
+    # === Parallelism ===
+    # Number of workers for collecting samples with. This only makes sense
+    # to increase if your environment is particularly slow to sample, or if
+    # you"re using the Async or Ape-X optimizers.
+    "num_workers": 0,
+    # Whether to compute priorities on workers.
+    "worker_side_prioritization": False,
+    # Prevent iterations from going lower than this time span
+    "min_iter_time_s": 1,
+
+    # === SlateQ specific options ===
+    # Learning method used by the slateq policy. Choose from: RANDOM,
+    # MYOP (myopic), SARSA, QL (Q-Learning),
+    "slateq_strategy": "QL",
+    # user/doc embedding size for the recsim environment
+    "recsim_embedding_size": 20,
+})
+# __sphinx_doc_end__
+# yapf: enable
+
+
+def validate_config(config: TrainerConfigDict) -> None:
+    """Checks the config based on settings"""
+    if config["slateq_strategy"] not in ALL_SLATEQ_STRATEGIES:
```

Comment:
Can we add a framework check here for `framework==torch`? Similar to DDPPO, QMIX, etc...

Reply:
Sure!